### DISCUSSION TITLE
extract to label

### DISCUSSION THREAD

**Discussion by mshanemc** [forcedotcom/salesforcedx-vscode#5771] [Ideas]
**Is your feature request related to a problem? Please describe.**
After coding something, converting hardcoded text to labels is a challenge.  You'd have to create them in the org UI, then pull them down, then cut/paste the correct API names into your code

people use labels if they want to 1) support i18n 2) let admins easily modify text without messing with code

**Describe the solution you'd like**
select some text in Apex/VF/LWC and have the extension turn it into a label in local source, replacing the text with the reference to a label (including, for LWCs, an import of the label in the js file).

**Describe alternatives you've considered**
copy paste all the things.

**Additional context**
For bonus points, detect if the label already exists, and rather than creating duplicate labels, it uses an existing one.

**Comment by smaddox-sf** [comment]
Thanks for submitting this @mshanemc
